migrations/helpers/update_filtered_positives_helper.py

```diff
@@ -161,36 +164,50 @@ def update_mongo_filtered_positive_fields(
     """
     with create_mongo_client(config) as client:
         mongo_db = get_mongo_db(config, client)
-
-        # get ids of those that are filtered positive, and those that aren't
-        all_ids: List[str] = [sample[FIELD_MONGODB_ID] for sample in samples]
-        filtered_positive_ids: List[str] = [
-            sample[FIELD_MONGODB_ID] for sample in list(filter(lambda x: x[FIELD_FILTERED_POSITIVE] is True, samples))
-        ]
-        filtered_negative_ids = [mongo_id for mongo_id in all_ids if mongo_id not in filtered_positive_ids]
-
         samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
-        samples_collection.update_many(
-            {FIELD_MONGODB_ID: {"$in": filtered_positive_ids}},
-            {
-                "$set": {
-                    FIELD_FILTERED_POSITIVE: True,
-                    FIELD_FILTERED_POSITIVE_VERSION: version,
-                    FIELD_FILTERED_POSITIVE_TIMESTAMP: update_timestamp,
-                }
-            },
-        )
-        samples_collection.update_many(
-            {FIELD_MONGODB_ID: {"$in": filtered_negative_ids}},
-            {
-                "$set": {
-                    FIELD_FILTERED_POSITIVE: False,
-                    FIELD_FILTERED_POSITIVE_VERSION: version,
-                    FIELD_FILTERED_POSITIVE_TIMESTAMP: update_timestamp,
-                }
-            },
-        )
-    return True
+
+        num_samples = len(samples)
+        SAMPLES_PER_QUERY = 15000
+        samples_index = 0
+        logger.debug(f"Attempting to update {num_samples} rows in Mongo in batches of {SAMPLES_PER_QUERY}")
+        while samples_index < num_samples:
```

Comment:
Would it be possible to add tests to check the batching of this? If existing tests pass that is a good sign, but they definitely don't input 15,000+ samples into this method 🙂

Reply:
This has been development-testing, and will be tested again during UAT testing